src/Engine/ProtoCore/Lang/Type.cs

```diff
@@ -417,7 +417,9 @@ public static StackValue Coerce(StackValue sv, Type targetType, RuntimeCore runt
                 return array.CopyArray(newTargetType, runtimeCore);
             }
 
-            if (!sv.IsArray && !sv.IsNull &&
+            // Null can be converted to Boolean so we will allow it in the case of indexable types
+            bool nullAsBool = sv.IsNull && (targetType.UID == (int)PrimitiveType.Bool);
+            if (!sv.IsArray && (!sv.IsNull || nullAsBool) &&
```

Comment:
this will allow `e :bool[]=null;` to be evaluated as `object[]{false}`

Reply:
Shouldn't this be `bool[]{false}`?

Reply:
Not sure how the VM stores the Booleans . 
The preview values are shown as `List { 0 False}`